pilot/pkg/model/service.go

```diff
@@ -60,14 +60,20 @@ type Service struct {
 	Mutex sync.RWMutex
 	// ClusterVIPs specifies the service address of the load balancer
 	// in each of the clusters where the service resides
-	ClusterVIPs map[string]string `json:"cluster-vips,omitempty"`
+	ClusterVIPs map[string]string `json:"clusterVips,omitempty"`
+
+	// ExternalAddresses is a mapping between a cluster name and the external
+	// address(es) to access the service from outside the cluster.
+	// Used by the aggregator to aggregate the Attributes.ExternalAddresses
+	// for clusters where the service resides
+	ExternalAddresses map[string][]string
```

Comment:
It doesn't matter what we cal this. This data structure here is an aggregation point. Its not the cleanest way to create this. Same applies to this ClusterVIPs that was created. Applicable to k8s, but it thoroughly confuses anybody else looking at the code to integrate it into their platform.

Move this into the service attribute, preferably under a map with key as k8s, within which you can do whatever you want, so that this logic does not leak into listeners/clusters, etc.

Also keep in mind that these new fields do not have a representation in the Service Entry. So what happens when Galley starts streaming service entries for all clusters? this will break badly. I suggest devising a more comprehensive solution.

Reply:
>when Galley starts streaming service entries for all clusters?

Is `ServiceEntry` the last choice? I donot think it is not easy to transform platform specific service model to ServiceEntry. If it is me, I would stream istio service model and thus reuse the convert logic.

Reply:
@rshriram  Is this still a blocker?

Reply:
Unfortunately, service entry is the format we are using to stream information from Galley to Pilot. So, you would have to figure out a way to make that work with service entry. May be through annotations, etc. and then translate into the service model. Hence my suggestion to stick this as attribute in service attributes

Reply:
ok, will take a look.